### PR TITLE
Migrated pkg/proxy/userspace to structured logging

### PR DESCRIPTION
**Pull request by shivanshuraj1333**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Migrate pkg/proxy/userspace to structured logging

#### Which issue(s) this PR fixes:

Fixes #
Ref https://github.com/kubernetes/kubernetes/issues/104872

#### Does this PR introduce a user-facing change?

```release-note
migrated pkg/proxy/userspace to structured logging
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/proxy/userspace/proxysocket.go [hunk 1]
```diff
@@ -95,18 +95,18 @@ func TryConnectEndpoints(service proxy.ServicePortName, srcAddr net.Addr, protoc
 	for _, dialTimeout := range EndpointDialTimeouts {
 		endpoint, err := loadBalancer.NextEndpoint(service, srcAddr, sessionAffinityReset)
 		if err != nil {
-			klog.Errorf("Couldn't find an endpoint for %s: %v", service, err)
+			klog.ErrorS(err, "Couldn't find an endpoint for service", "service", service)
 			return nil, err
 		}
-		klog.V(3).Infof("Mapped service %q to endpoint %s", service, endpoint)
+		klog.V(3).InfoS("Mapped service to endpoint", "serivce", service, "endpoint", endpoint)
```
Comment (thockin) [inline pkg/proxy/userspace/proxysocket.go:101]: misspelled "service'

Comment (shivanshu1333) [reply pkg/proxy/userspace/proxysocket.go]: Ahh, my bad, thanks for pointing out, fixed

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by pacoxu** [comment]
/hold
The CI failure should be fixed by running `./hack/update-gofmt.sh`.

BTW, `fix typo`can be squashed with the migrating commit for nit.

**Comment by shivanshu1333** [comment]
Work done after receiving `lgtm` label:
1) Squashed `typo fix` commit
2) code formatting done using `./hack/update-gofmt.sh`

**Comment by shivanshu1333** [comment]
@thockin can you please re add the `lgtm` label?

> Work done after receiving lgtm label:
> Squashed typo fix commit
> code formatting done using ./hack/update-gofmt.sh

There is no additional change, thanks!

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**